src/query/sql/src/planner/optimizer/heuristic/prune_unused_columns.rs

```diff
@@ -156,10 +156,24 @@ impl UnusedColumnPruner {
                     Self::keep_required_columns(expr.child(0)?, required)?,
                 ))
             }
-            RelOperator::Window(p) => Ok(SExpr::create_unary(
-                RelOperator::Window(p.clone()),
-                Self::keep_required_columns(expr.child(0)?, required)?,
-            )),
+            RelOperator::Window(p) => {
+                p.aggregate_args.iter().for_each(|arg| {
+                    required.insert(arg.index);
+                });
+
+                p.partition_by.iter().for_each(|item| {
+                    required.insert(item.index);
+                });
+
+                p.order_by.iter().for_each(|item| {
+                    required.insert(item.order_by_item.index);
+                });
```

Comment:
Should insert inner columns of scalar items (Refer to the codes for `RelOperator::Aggregate` above.)